### ISSUE TITLE
Node Resizer: add option to overwrite the resize direction

### ISSUE THREAD

**Issue by chrtze** [xyflow/xyflow#2986]
### Describe the Bug

Currently, the controls of the NodeResizer get their possible resize directions based on their positions. This means that it's not possible to limit a control in the bottom-right corner to resize only horizontally. Maybe we can add an option to overwrite the direction in which a control can resize the node.

### Your Example Website or App

https://codesandbox.io/s/zen-wind-w9znpf?file=/ResizableNode.js

### Steps to Reproduce the Bug or Issue

* Add a node resize control with variant `Handle` to your custom node
* Add a shouldResize function that only returns true for one resize direction
* This does work with variant="Line" but not with variant="Handle"

### Expected behavior

It should be possible to limit the direction for all controls

### Platform

all

### Additional context

https://github.com/wbkd/react-flow/discussions/2868

**Comment by jt6677** [comment]
shouldResize={(e, params) => params.direction[1] === 0}      does not work on diagonal.

**Comment by peterkogo** [comment]
Good idea. We will add an additional prop to NodeResizeControl so the displayed position can differ from the resize direction.

**Comment by Praneeth-Pike** [comment]
Hey @peterkogo any update on this issue? I need to only be able to resize node width, and leave height as is.

**Comment by chrtze** [comment]
I looked into this and think that this would be a good improvement. At the same time, we could adjust the logic for one-directional resize controls to not set the `width` or `height` if it doesn't change. This would fix the issue that @Praneeth-Pike was mentioning I guess. This would enable users to resize horizontally while keeping the height of the node dynamic.

https://github.com/user-attachments/assets/0dbc542f-4835-4506-86b7-51dd8e3f8018

**Comment by chrtze** [comment]
Proposal for solving these issues:

- make both `width` and `height` optional in a dimensions change object (currently both are required, so it's not possible to set only one of them)
- only apply the dimensions that are set in the object
- add a property to the node resize control and the node resizer (for example: direction="both | horizontal | vertical") and only add width / height to the dimensions change when horizontal or vertical is set

**Comment by moklick** [comment]
In 12.6.0 you can specify an optional `resizeDirection`('horzontal' or 'vertical') for a `NodeResizeControl` component: https://reactflow.dev/api-reference/components/node-resize-control#resizedirection
